File: vuln-sandbox/python/taint/cmdi-1hop.py
# Taint: Command Injection — 1-hop (Python)
# Source: request.args.get('cmd') → Sink: os.system
# Expected: TruePositive (taint/cwe78)
from flask import Flask, request
import os

app = Flask(__name__)

@app.route('/run')
def run_command():
    cmd = request.args.get('cmd')              # SOURCE: HTTP request param
    os.system(f"ls {cmd}")                     # SINK: shell command
    return 'done'

File: vuln-sandbox/node/taint/cmdi-1hop.js
// Taint: Command Injection — 1-hop (JS)
// Source: req.query.cmd → Sink: child_process.exec
// Expected: TruePositive (taint/cwe78)
const express = require('express');
const { exec } = require('child_process');
const app = express();

app.get('/run', (req, res) => {
    const cmd = req.query.cmd;                  // SOURCE: HTTP request param
    exec(`ls ${cmd}`, (err, stdout) => {        // SINK: shell command
        res.send(stdout);
    });
});
